jdaviz/configs/specviz/plugins/unit_conversion/unit_conversion.py

```diff
@@ -461,19 +462,25 @@ def _handle_attribute_display_unit(self, attr_unit, layers=None):
 
         for layer in layers:
             # DQ layer doesn't play nicely with this attribute
+            # TODO: detect a DQ layer in a way that doesn't depend on the label
+            if not hasattr(layer.state, 'attribute_display_unit'):
+                continue
             if "DQ" in layer.layer.label or isinstance(layer.layer, GroupedSubset):
                 continue
-            elif ("flux" not in [str(c) for c in layer.layer.components]
-                    or u.Unit(layer.layer.get_component("flux").units).physical_type != 'surface brightness'):  # noqa
+            for comp in layer.layer.components:
+                if u.Unit(layer.layer.get_component(comp).units).physical_type == 'surface brightness':  # noqa
+                    break
+            else:
+                # if no component with surface brightness units, skip this layer
                 continue
```

Comment:
The logic in these lines is quite confusing... I'm thinking it'd be more clear (albeit technically/possibly less performant) to use a list comprehension to check for 'surface brightness' in each layer.

```suggestion
            has_surface_brightness = len([True for comp in layer.layer.components if u.Unit(layer.layer.get_component(comp).units).physical_type == 'surface brightness']) # noqa
                                                                 
            # if no component with surface brightness units, skip this layer
            if not has_surface_brightness:
                continue
```

Reply:
this definitely will be less performant since the break will stop as soon as it finds the first one instead of needing to loop through all and store in a list.  Is there a comment that can be added to make it less confusing?

Reply:
I just realized the best of both worlds might be to have a separate function to run that for loop (+ ``break``) and use the function as the ``if`` condition.

Reply:
I love for-else!  I like the function idea though, let me see if I can get that in here